### PR TITLE
remove ENABLE_TELEMETRY macro

### PR DESCRIPTION
**Pull request by zhangxiang1993**
**Description**: This PR is to remove the ENABLE_TELEMETRY macro added in the generated traceloggingconfigprivate.h, which can be replaced by the TraceLoggingOptionMicrosoftTelemetry itself, so to reduce the code needed to be generated.

### CODE REVIEW

**Comment by snnn** [comment]
Why? Could you please show us more detail?

**Comment by zhangxiang1993** [comment]
> Why? Could you please show us more detail?

I edited the description. This PR will not affect the telemetry behavior of Nuget Package, it's just a little refactor, because we have some corresponding change in WindowsAI to bring back the telemetry of our inbox dlls.

File: include/onnxruntime/core/platform/windows/TraceLoggingConfig.h [hunk 1]
```diff
@@ -23,7 +23,7 @@ Module Name:
 
 // Configuration macro for use in TRACELOGGING_DEFINE_PROVIDER. The definition
 // in this file configures the provider as a normal (non-telemetry) provider.
-#ifndef ENABLE_TELEMETRY
+#ifndef TraceLoggingOptionMicrosoftTelemetry
```
Comment (snnn) [inline include/onnxruntime/core/platform/windows/TraceLoggingConfig.h:26]: I understand this macro can also be defined somewhere else, not TraceLoggingConfigPrivate.h. But in that case are you able to ensure that header file is always included before this TraceLoggingConfig.h  file?

Comment (tiagoshibata) [reply include/onnxruntime/core/platform/windows/TraceLoggingConfig.h:26]: It should work if it's included after (it'll override the old value)

Comment (tiagoshibata) [reply include/onnxruntime/core/platform/windows/TraceLoggingConfig.h:26]: Though it might raise a warning for redefining the macro

Comment (snnn) [reply include/onnxruntime/core/platform/windows/TraceLoggingConfig.h:26]: Fine. That's so tricky. :-)

**Pull request merged**